tests/cbmc/proofs/s2n_hex_string_to_bytes/s2n_hex_string_to_bytes_harness.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "error/s2n_errno.h"
+#include "utils/s2n_blob.h"
+
+#include <assert.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <string.h>
+
+void s2n_hex_string_to_bytes_harness()
+{
+    /* Non-deterministic inputs. */
+    struct s2n_blob *blob = cbmc_allocate_s2n_blob();
+    __CPROVER_assume(s2n_blob_is_valid(blob));
+    char *str = ensure_c_str_is_allocated(MAX_STRING_LEN);
+
+    /* Save previous state. */
+    struct s2n_blob old_blob = *blob;
+
+    /* Operation under verification. */
+    if(s2n_hex_string_to_bytes(str, blob) == S2N_SUCCESS) {
+        size_t strLength = strlen(str);
+        assert(blob->size >= (strLength / 2));
```

Comment:
isn't the encoding size exactly half?

Reply:
The `s2n_hex_string_to_bytes` function only checks for
```c
   /* protects against overflows */
    gte_check(blob->size, len / 2);
```
So there is a path where `blob->size` will be greater than `len/2`.